File: camerapage.cpp
#include "camerapage.h"
#include "ui_camerapage.h"
#include "telemetrydata.h"

CameraPage::CameraPage(QWidget* parent)
    : QWidget(parent), ui(new Ui::CameraPage)
{
    ui->setupUi(this);

    connect(ui->btnRear, &QPushButton::clicked, this, [this](){ setMode("REAR"); });
    connect(ui->btnFront, &QPushButton::clicked, this, [this](){ setMode("FRONT"); });
    connect(ui->btnMosaic, &QPushButton::clicked, this, [this](){ setMode("MOSAIC"); });
    connect(ui->btnBird, &QPushButton::clicked, this, [this](){ setMode("BIRD-EYE"); });
}

CameraPage::~CameraPage(){ delete ui; }

void CameraPage::setMode(const QString& mode)
{
    ui->lblMode->setText(QString("Mode: %1").arg(mode));
}

void CameraPage::bindTelemetry(TelemetryData* t)
{
    m_t = t;
    if(!m_t) return;

    auto refresh = [this](){
        // petit feedback visuel dans le placeholder si reverse
        if(m_t->reverse()){
            ui->lblVideo->setText("Flux caméra / Bird-eye - placeholder\n(REVERSE engagé)");
            ui->lblVideo->setStyleSheet("QLabel{color:#FFD27C; font-size:16px;}");
        } else {
            ui->lblVideo->setText("Flux caméra / Bird-eye - placeholder");
            ui->lblVideo->setStyleSheet("QLabel{color:#b8c0cc; font-size:16px;}");
        }
    };
    refresh();
    connect(m_t, &TelemetryData::reverseChanged, this, refresh);
}

File: camera_page.py
from __future__ import annotations

from PyQt6 import uic
from PyQt6.QtWidgets import QWidget

from telemetry_data import TelemetryData


class CameraPage(QWidget):
    def __init__(self, parent: QWidget | None = None) -> None:
        super().__init__(parent)
        uic.loadUi("camerapage.ui", self)
        self._telemetry: TelemetryData | None = None

        self.btnRear.clicked.connect(lambda: self.set_mode("REAR"))
        self.btnFront.clicked.connect(lambda: self.set_mode("FRONT"))
        self.btnMosaic.clicked.connect(lambda: self.set_mode("MOSAIC"))
        self.btnBird.clicked.connect(lambda: self.set_mode("BIRD-EYE"))

    def set_mode(self, mode: str) -> None:
        self.lblMode.setText(f"Mode: {mode}")

    def bind_telemetry(self, telemetry: TelemetryData) -> None:
        self._telemetry = telemetry
        if self._telemetry is None:
            return

        refresh = self._refresh
        refresh()
        self._telemetry.reverseChanged.connect(refresh)

    def _refresh(self) -> None:
        if self._telemetry is None:
            return

        if self._telemetry.reverse:
            self.lblVideo.setText("Flux caméra / Bird-eye - placeholder\n(REVERSE engagé)")
            self.lblVideo.setStyleSheet("QLabel{color:#FFD27C; font-size:16px;}")
        else:
            self.lblVideo.setText("Flux caméra / Bird-eye - placeholder")
            self.lblVideo.setStyleSheet("QLabel{color:#b8c0cc; font-size:16px;}")
